src/components/SearchBar/SearchBar.component.tsx

```diff
@@ -0,0 +1,95 @@
+import type { FC } from 'react';
+import { useState } from 'react';
+
+import { ArrowDown } from '@/assets/icons/ArrowDown.tsx';
+import { images } from '@/constants/images.constant.ts';
+import type { Themes } from '@/types/themes.type.ts';
+
+import styles from './SearchBar.module.css';
+
+const categories = ['Electronics', 'Shoes', 'Clothes'] as const;
+
+const sortOptions = ['Price (Low-High)', 'Newest', 'Oldest'] as const;
+
+interface SearchBarProps {
+    theme: Themes;
+}
+
+const SearchBar: FC<SearchBarProps> = ({ theme }) => {
+    const [searchQuery, setSearchQuery] = useState<string>('');
+    const [selectedCategory, setSelectedCategory] = useState<string>();
+    const [sortOption, setSortOption] = useState<string>(sortOptions[0]);
+    const [isSortOptionsOpen, setIsSortOptionsOpen] = useState<boolean>(false);
+
+    // const handleSearch = () => {
+    // handle search
+    // };
+
+    const handleCategoryFilter = (category: string) => {
+        setSelectedCategory(category);
+    };
+
+    const handleSortOptionsOpen = (option: boolean) => {
+        setIsSortOptionsOpen(option);
+    };
+
+    const handleSort = (option: string) => {
+        setSortOption(option);
+    };
+
+    return (
+        <div className={styles.searchBar}>
+            <form className={styles.searchForm}>
```

Comment:
Could you consider splitting this component into 3 smaller parts:
- `searchBar`
- `productsCategories`
-  `priceCategories`

Naming is up to you